lib/spack/spack/test/cmd/gpg.py

```diff
@@ -48,6 +49,9 @@ def test_find_gpg(cmd_name, version, tmpdir, mock_gnupghome, monkeypatch):
 
 def test_no_gpg_in_path(tmpdir, mock_gnupghome, monkeypatch):
     monkeypatch.setitem(os.environ, "PATH", str(tmpdir))
+    monkeypatch.setattr(
+        spack.bootstrap, 'ensure_gpg_in_path_or_raise', lambda: None
+    )
```

Comment:
Wouldn't it be better to check this by "untrusting" all the bootstrapping methods? That way we're actually checking the proper behavior if gpg isn't available, rather than testing what happens if bootstrapping fails.